mitodl/mit-open
# Adds ETL to pull PostHog view events into the database; adds popular resource APIs

### What are the relevant tickets?

Closes mitodl/hq#4008

### Description (What does it do?)

Two main things:
- This adds a pipeline to the normal learning resources ETL pipeline stack that pulls the learning resource view events (`lrd_view`) from PostHog, and stores them in the local database.
- This also adds a `popular` endpoint to some learning resource APIs, which will pull resources in order of popularity (most hits to least). This is done in the same way that new and upcoming resources are done.

### How can this be tested?

Automated tests should pass.

To test you will need to have PostHog set up and enabled - see the README.md for this. Note that you will also need a personal API key for this to work, so if you had this set up before you will need to add this setting. It's new in this PR. You will also need to generate some views, and you'll need to pull the view data into the system. 

_Testing the pipeline manually_: You can run the `load_posthog_lrd_view_events` management command to run the pipeline manually. 
_Testing the task-based pipeline runner_: The task is called `get_learning_resource_views` and you should be able to queue it up and it should run.
_Testing the endpoints_: You should be able to append `popular` to the `learning_resources`, `courses`, and `programs` endpoints. This will return a list of the given resource sorted by the number of views each has. (I find the easiest way to test this is to load the API directly in a browser that has JSON View or something similar, or on the command line piped into `jq`.)
